components/script/dom/location.rs

```diff
@@ -191,14 +248,20 @@ impl LocationMethods for Location {
 
     // https://html.spec.whatwg.org/multipage/#dom-location-href
     fn SetHref(&self, value: USVString) -> ErrorResult {
-        // Note: no call to self.check_same_origin_domain()
-        let current_url = self.window.get_url();
-        let url = match current_url.join(&value.0) {
-            Ok(url) => url,
-            Err(e) => return Err(Error::Type(format!("Couldn't parse URL: {}", e))),
-        };
-        let referrer = Referrer::ReferrerUrl(current_url.clone());
-        self.navigate(url, referrer, HistoryEntryReplacement::Disabled, false);
+        // Step 1: If this Location object's relevant Document is null, then return.
+        if self.window.has_document() {
+            // Note: no call to self.check_same_origin_domain()
+            // Step 2: Parse the given value relative to the entry settings object.
+            // If that failed, throw a TypeError exception.
+            let base_url = self.entry_document().url();
+            let url = match base_url.join(&value.0) {
+                Ok(url) => url,
+                Err(e) => return Err(Error::Type(format!("Couldn't parse URL: {}", e))),
+            };
+            // Step 3: Location-object-setter navigate to the resulting URL record.
+            let referrer = Referrer::ReferrerUrl(base_url.clone());
```

Comment:
Ok I think this is the problem with `/html/browsers/browsing-the-web/navigating-across-documents/source/navigate-child-function.html`, the referrer should be not from the entry document but from `self.window.Document`, at least that is what I make from the test failure(so it's not a new intermittent and not actually related much to navigation)

Reply:
Ahh, right. The referrer url shouldn't be from entry document here - I think `let referrer = Referrer::ReferrerUrl(self.get_url());` should fix this.